### DISCUSSION TITLE
Does withConverter need to have fromFirestore and toFirestore?

### DISCUSSION THREAD

**Discussion by joshpetit** [firebase/flutterfire#7707] [Q&A] [answered]
On the client side I'm only ever reading documents, so I was wondering if it was really necessary to make the `toFirestore` field required on `withConverter` for documents? Shouldn't it only be required when you're writing data? I'd supposed the same for the other way around, you shouldn't really need to define a `fromFirestore` if you're only writing data.

**Comment by rrousselGit** [comment, marked as answer]
If you don't want to implement `toFirestore`, you're free to throw an exception

```dart
toFirestore: (value, options) => throw UnsupportedError(),
```

**Reply by joshpetit** [reply]
Amazing thanks, is this in the docs already? If not would be a great addition!
